c7n/resources/s3.py

```diff
@@ -2762,6 +2762,288 @@ def get_inventory_delta(self, client, inventory, b):
+                                },
+                                'And': {
+                                    'type': 'object',
+                                    'additionalProperties': False,
+                                    'properties': {
+                                        'Prefix': {'type': 'string'},
+                                        'Tags': {
+                                            'type': 'array',
+                                            'items': {
+                                                'type': 'object',
+                                                'required': ['Key', 'Value'],
+                                                'additionalProperties': False,
+                                                'properties': {
+                                                    'Key': {'type': 'string'},
+                                                    'Value': {'type': 'string'},
+                                                },
+                                            },
+                                        },
+                                    },
+                                },
+                            },
+                        },
+                        'Tierings': {
+                            'type': 'array',
+                            'items': {
+                                'type': 'object',
+                                'additionalProperties': False,
+                                'properties': {
+                                    'Days': {'type': 'integer'},
+                                    'AccessTier': {'type': 'string', 'enum': [
+                                        'ARCHIVE_ACCESS', 'DEEP_ARCHIVE_ACCESS']},
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    }
+
+    permissions = ('s3:PutIntelligentTieringConfiguration',)
+
+    def validate(self):
+        # You can have up to 1,000 S3 Intelligent-Tiering configurations per bucket.
+        # Hence, always use it with a filter
+        for f in self.manager.iter_filters():
+            if isinstance(f, IntelligentTiering):
+                return self
+        raise PolicyValidationError(
+            '`set-intelligent-tiering` may only be used in '
+            'conjunction with `intelligent-tiering` filter on %s' % (self.manager.data,))
+
+    def process(self, buckets):
+        with self.executor_factory(max_workers=3) as w:
+            futures = {}
+            results = []
+
+            for b in buckets:
+                futures[w.submit(self.process_bucket, b)] = b
+
+            for future in as_completed(futures):
+                if future.exception():
+                    bucket = futures[future]
+                    self.log.error('error modifying bucket lifecycle: %s\n%s',
+                                   bucket['Name'], future.exception())
```

Comment:
missing a continue in this block, else future.result() below raises

Reply:
Addressed